src/current/v25.2/vector-indexes.md

```diff
@@ -0,0 +1,323 @@
+---
+title: Vector Indexes
+summary: How CockroachDB uses vector indexes to efficiently perform approximate nearest neighbor (ANN) searches on high-dimensional vector data.
+toc: true
+keywords: vector, vector index, vector search, ann, approximate nearest neighbor, inverted index, accelerated index
+docs_area: develop
+---
+
+{{site.data.alerts.callout_info}}
+{% include_cached feature-phases/preview.md %}
+{{site.data.alerts.end}}
+
+{% include_cached new-in.html version="v25.2" %} A *vector index* enables efficient approximate nearest neighbor (ANN) search on high-dimensional [`VECTOR`]({% link {{ page.version.version }}/vector.md %}) columns. Use vector indexes to improve the performance of similarity searches over large datasets, such as embeddings generated by machine learning models.
+
+This page describes how to create and use vector indexes on CockroachDB.
+
+{{site.data.alerts.callout_info}}
+`VECTOR` functionality is compatible with the [`pgvector`](https://github.com/pgvector/pgvector) extension for PostgreSQL.
+{{site.data.alerts.end}}
+
+## How do vector indexes work?
+
+CockroachDB vector indexes organize [vectors]({% link {{ page.version.version }}/vector.md %}) into a hierarchical structure of partitions using [k-means clustering](https://en.wikipedia.org/wiki/K-means_clustering). This partition structure groups similar vectors together and enables efficient, [tunable](#tune-vector-indexes) ANN searches.
+
+When a query uses a vector index, CockroachDB explores a subset of partitions based on their proximity to the query vector. It then retrieves and evaluates a candidate set of vectors using the [configured distance metric](#comparisons) and returns the top nearest results.
+
+## Enable vector indexes
+
+To enable the use of vector indexes, set the `feature.vector_index.enabled` [cluster setting]({% link {{ page.version.version }}/cluster-settings.md %}#setting-feature-vector-index-enabled):
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+SET CLUSTER SETTING feature.vector_index.enabled = true;
+~~~
+
+To enable the creation of vector indexes on non-empty tables, also disable the `sql_safe_updates` [session setting]({% link {{ page.version.version }}/session-variables.md %}#sql-safe-updates). This allows vector indexes to be backfilled on existing rows, during which **table writes are blocked** to ensure vector index consistency.
```

Comment:
Should we mention that this is a temporary limitation during the Preview, and will be lifted in a future release?

Reply:
Yeah I think we should.

Reply:
Which part is the temporary limitation -- the gating behind `sql_safe_updates`, or the table writes being blocked?

Reply:
Both - we gated non-empty vector index backfill behind that setting because of this blocking behavior that we plan to lift.